koku/koku/trino_database.py

```diff
@@ -149,10 +241,13 @@ def executescript(trino_conn, sqlscript, *, params=None, preprocessor=None):
                 cur.execute(stmt, params=s_params)
                 results = cur.fetchall()
             except TrinoQueryError as trino_exc:
+                LOG.warning(f"{trino_exc!s}")
+                if "NoSuchKey" in str(trino_exc):
+                    raise TrinoNoSuchKeyException
+
                 trino_statement_error = TrinoStatementExecError(
                     statement=stmt, statement_number=stmt_num, sql_params=s_params, trino_error=trino_exc
                 )
-                LOG.warning(f"{trino_statement_error!s}")
                 raise trino_statement_error from trino_exc
```

Comment:
This can be reworked to log the `str` of the exception being raised and pass some parameters to the `TrinoNoSuchKeyException` to provide better conetxt in the logs.
```python
                exc_to_raise = TrinoStatementExecError(
                    statement=stmt, statement_number=stmt_num, sql_params=s_params, trino_error=trino_exc
                )
                if "NoSuchKey" in str(trino_exc):
                    exc_to_raise = TrinoNoSuchKeyException(
                        message=trino_exc.message,
                        query_id=trino_exc.query_id,
                        error_code=trino_exc.error_code,
                    )

                LOG.warning(f"{exc_to_raise!s}")
                raise exc_to_raise from trino_exc
```